### PR TITLE
log-backup: fix the issue that gc-worker owner cannot check log-backup task existed.

### PR DESCRIPTION
**Pull request by joccau**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/41806
close https://github.com/pingcap/tidb/issues/41642

Problem Summary:
If the gc-worker owner and the pitr owner is not the same node, the gc-worker owner cannot check the log-backup task existed.

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

## Manual test
- Create a TiDB cluster with the 3 tidb-server, 3 pd-server and 3 tikv-server.
- Create a log-backup task, Search the log in all of tidb-server, the log like this
```
2023-03-01 16:38:04 (UTC+08:00)TiDB 10.2.5.20:4100[db.go:206] ["inc log backup task"] [count=1]
2023-03-01 16:38:07 (UTC+08:00)TiDB 10.2.5.31:4100[db.go:206] ["inc log backup task"] [count=1]
2023-03-01 16:38:10 (UTC+08:00)TiDB 10.2.5.32:4100[db.go:206] ["inc log backup task"] [count=1]
```
- restart all of tidb-server, and search the log
```
2023-03-01 17:11:27 (UTC+08:00)TiDB 10.2.5.31:4100[db.go:206] ["inc log backup task"] [count=1]
2023-03-01 17:11:27 (UTC+08:00)TiDB 10.2.5.32:4100[db.go:206] ["inc log backup task"] [count=1]
2023-03-01 17:11:27 (UTC+08:00)TiDB 10.2.5.20:4100[db.go:206] ["inc log backup task"] [count=1]
```

- stop a log-backup task. Search the log
```
2023-03-01 17:13:02 (UTC+08:00)TiDB 10.2.5.32:4100[db.go:212] ["dec log backup task"] [count=0]
2023-03-01 17:13:02 (UTC+08:00)TiDB 10.2.5.20:4100[db.go:212] ["dec log backup task"] [count=0]
2023-03-01 17:13:02 (UTC+08:00)TiDB 10.2.5.31:4100[db.go:212] ["dec log backup task"] [count=0]
```
- start the log-backup task again, and search the log
```
2023-03-01 17:13:50 (UTC+08:00)TiDB 10.2.5.32:4100[db.go:206] ["inc log backup task"] [count=1]
2023-03-01 17:13:50 (UTC+08:00)TiDB 10.2.5.31:4100[db.go:206] ["inc log backup task"] [count=1]
2023-03-01 17:13:50 (UTC+08:00)TiDB 10.2.5.20:4100[db.go:206] ["inc log backup task"] [count=1]
```

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
fix the issue that gc owner cannot check log-backup task existed
```

### CODE REVIEW

File: br/pkg/streamhelper/daemon/owner_daemon_test.go [hunk 1]
```diff
@@ -33,9 +33,12 @@ func newTestApp(t *testing.T) *anApp {
 	}
 }
 
+func (a *anApp) OnStart(ctx context.Context) {
+}
+
 // OnStart would be called once become the owner.
```
Comment (tonyxuqqi) [inline br/pkg/streamhelper/daemon/owner_daemon_test.go:39]: comment "OnStart" needs to be replaced.

**Review by tonyxuqqi — APPROVED** [review]

**Review by 3pointer — APPROVED** [review]

**Review by YuJuncen — APPROVED** [review]

**Pull request merged**